### PR TITLE
Add medium bold to header

### PR DESCRIPTION
**Pull request by shaharzil**
https://monday.monday.com/boards/3532714909/pulses/6899636911

### CODE REVIEW

File: packages/core/src/components/EditableHeading/__stories__/EditableHeading.stories.module.scss [hunk 1]
```diff
@@ -12,5 +12,9 @@
     .editableHeading {
       width: 200px;
     }
+
+    .editableHeading:first-child {
+      width: 125px;
```
Comment (talkor) [inline packages/core/src/components/EditableHeading/__stories__/EditableHeading.stories.module.scss:17]: Why this change?

File: packages/style/src/config/tokens/core/_core-typography.scss [hunk 2]
```diff
@@ -10,12 +10,15 @@ $title-font-family: $font-family-english-heading, $font-family-other-languages;
...
 $font-h1-bold: 700 32px / 40px var(--title-font-family);
+$font-h1-medium: 600 32px / 40px var(--title-font-family);
 $font-h1-normal: 500 32px / 40px var(--title-font-family);
 $font-h1-light: 300 32px / 40px var(--title-font-family);
 $font-h2-bold: 700 24px / 30px var(--title-font-family);
+$font-h2-medium: 600 24px / 30px var(--title-font-family);
 $font-h2-normal: 500 24px / 30px var(--title-font-family);
 $font-h2-light: 300 24px / 30px var(--title-font-family);
-$font-h3-bold: 600 18px / 24px var(--title-font-family);
+$font-h3-bold: 700 18px / 24px var(--title-font-family);
```
Comment (talkor) [inline packages/style/src/config/tokens/core/_core-typography.scss:20]: I know it's a really minor change but are we ok with that?

Comment (shaharzil) [reply packages/style/src/config/tokens/core/_core-typography.scss:20]: I think we are :)

File: packages/core/src/components/EditableHeading/__stories__/EditableHeading.stories.module.scss (hunk 1 above)
Comment (talkor) [reply packages/core/src/components/EditableHeading/__stories__/EditableHeading.stories.module.scss:17]: @shaharzil

Comment (shaharzil) [reply packages/core/src/components/EditableHeading/__stories__/EditableHeading.stories.module.scss:17]: Because it gets really ugly with the addition of a fourth category

**Review by talkor — APPROVED** [review]
